queries.sql

```diff
@@ -31,3 +31,122 @@ WHERE name != 'Gabumon';
+*/
+-- Start a transaction
+BEGIN;
+
+-- Update the "species" column to "unspecified"
+UPDATE animals SET species = 'unspecified';
+
+-- Verify that the change was made (Check the "species" column)
+SELECT * FROM animals;
+
+-- Roll back the transaction
+ROLLBACK;
+
+-- Verify that the "species" column returns to its original state
+SELECT * FROM animals;
+
+-----------------------------------------------
+
+/*
+  * This code begins a transaction, updates the "species" column
+    1. to "digimon" for all animals that have a name ending in "mon"
+    2. to "pokemon" for all animals that don't have a species already set.
+
+  * This code verifies changes within the transaction, commits the changes, and verifies that the changes persist after the commit.
+*/
+-- Start a transaction
+BEGIN;
+
+-- Update "species" to "digimon" for animals with names ending in "mon"
+UPDATE animals SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+-- Update "species" to "pokemon" for animals without a species set
+UPDATE animals SET species = 'pokemon'
+WHERE species IS NULL;
+
+-- Verify changes within the transaction
+SELECT * FROM animals;
+
+-- Commit the transaction
+COMMIT;
+
+-- Verify that changes persist after the commit
+SELECT * FROM animals;
+
+-----------------------------------------------
+
+/*
+  * This code begins a transaction,
+  * deletes all records in the "animals" table,
+  * rolls back the transaction,
+  * verifies if all records in the "animals" table still exist.
+*/
+
+-- Start a transaction
+BEGIN;
+
+-- Delete all records in the "animals" table
+DELETE FROM animals;
+
+-- Roll back the transaction
+ROLLBACK;
+
+-- Verify if all records in the "animals" table still exist
+SELECT * FROM animals;
+
+-----------------------------------------------
+
+/*
+  * The following code begins a transaction,
+  * deletes all records of animals born after Jan 1st, 2022.
+  * creates a savepoint for the transaction,
+  * updates all animals' weight to be their weight multiplied by -1,
+  * rolls back to the savepoint,
+  * updates all animals' weights that are negative to be their weight multiplied by -1,
+  * commits the transaction.
+*/
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
+-----------------------------------------------
+
+-- Find number of animals in the table
+SELECT COUNT(*) AS total_animals FROM animals;
+
+-- Find number of animals that have never tried to escape
+SELECT COUNT(*) AS never_tried_to_escape FROM animals
+WHERE escape_attempts = 0;
+
+-- Find the average weight of animals
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+-- Animal who escapes the most, neutered or not neutered animals
+SELECT name, MAX(escape_attempts) AS max_escape_attempts
+FROM animals
+GROUP BY name;
```

Comment:
You did well with this project, kindly make few changes with this query as stated below, the query is to fid out either `neutered` or ` not neutered` :

`  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;
`

Reply:
Noted @usorfaitheloho.
Thank you for you review and feedback.